source/common/http/http1/codec_impl.cc

```diff
@@ -407,34 +413,54 @@ void RequestEncoderImpl::encodeHeaders(const RequestHeaderMap& headers, bool end
   encodeHeadersBase(headers, absl::nullopt, end_stream);
 }
 
+int ConnectionImpl::setAndCheckCallbackStatus(Status&& status) {
+  codec_status_ = std::move(status);
```

Comment:
Are there some invariants on codec_status_ that should be checked here?

I think at least: ASSERT(codec_status_.ok()) prior to overwriting.

Reply:
@yanavlasov for this change in H/2